src/topology/config/watcher.rs

```diff
@@ -0,0 +1,113 @@
+use super::Config;
+use futures::{Async, Future, Stream};
+use notify::{watcher, DebouncedEvent, RecursiveMode, Watcher};
+use std::{
+    path::Path,
+    sync::mpsc::{channel, TryRecvError},
+    time::{Duration, Instant},
+};
+use tokio::timer::Delay;
+
+// Per notify own documentation, it's advised to have delay of more than 30 sec,
```

Comment:
Leaving a note: We should ensure this is documented but I'd also like to make sure the rest of the team is aware of this.

Reply:
> Per notify own documentation, it's advised to have delay of more than 30 sec so to avoid receiving repetitions of previous events on macOS.

What if the `notify` dependency was upgraded to 0.5.0-pre.2 and [immediate_watcher](https://docs.rs/notify/5.0.0-pre.2/notify/fn.immediate_watcher.html) was used instead, without any delay?

Reply:
That would work, but would be fragile. There is a slew of platform X editor details, and concurrency issues, that are largely mitigated with a delay. While without it, config update logic would have to deal with it.